pkg/drivers/common/vmnet/vmnet.go

```diff
@@ -72,6 +72,14 @@ type interfaceInfo struct {
 // The returned error.Kind can be used to provide a suggestion for resolving the
 // issue.
 func ValidateHelper() error {
+	// Ideally minikube will not try to validate in download-only mode, but this
+	// is called from different places in different drivers, so the easier way
+	// to skip validation is to skip it here.
+	if viper.GetBool("download-only") {
```

Comment:
I have been trying to discourge checking the viper config in the packages deep inside. the guideline is all the viper stuff needs to be in cmd, how about make an optional arg for this func that if passed Skip to skip it.

Reply:
it should be in https://github.com/kubernetes/minikube/blob/71227b25cd999e4333dc9e85b986d6fb50b350cf/cmd/minikube/cmd/start_flags.go#L538

Reply:
We already use viper for checking if we run with --interactive=false, so using download-only matches the current code.

The reason validateVfkitNetwork() is not used for krunkit is that it does not have a --network flag so there is nothing to validate. In krunkit case we validate in the driver Startus(), same we validate the kvm driver configuration.

I can improve it later to pass options to validate, since the interactive check must be done in validate.

Reply:
interactive flag is not as bad as "download-only flag" since that one is used more in the test. I think since we are already here, we should move all the viper ones here to cmd packages